invertedai/api/blame.py

```diff
@@ -0,0 +1,161 @@
+import time
+from typing import List, Optional, Tuple, Dict
+from pydantic import BaseModel, validate_arguments
+
+import sys
+sys.path.append("..")
+
+import invertedai as iai
+from invertedai.api.config import TIMEOUT, should_use_mock_api
+from invertedai.api.mock import (
+    get_mock_birdview,
+    get_mock_blamed_result,
+    get_mock_blamed_reasons,
+    get_mock_confidence_score,
+)
+from invertedai.error import APIConnectionError, InvalidInput
+from invertedai.common import (
+    AgentState,
+    Image,
+    AgentAttributes,
+    TrafficLightStatesDict,
+)
+
+
+class BlameResponse(BaseModel):
+    """
+    Response returned from an API call to :func:`iai.blame`.
+    """
+    blamed_collisions: Optional[List[Tuple[Tuple[int, int], Tuple[int, int], Tuple[int, ...]]]]
+    blamed_result: Optional[Tuple[int, ...]]
+    reasons: Optional[Dict[int, List[str]]]
+    confidence_score: Optional[float]
+    birdviews: Optional[List[Image]]  #: If `get_birdviews` was set, this contains the resulting image.
+
+
+@validate_arguments
+def blame(
+    location: str,
+    candidate_agents: Tuple[int, int],
+    agent_state_history: List[List[AgentState]],
+    agent_attributes: List[AgentAttributes],
+    traffic_light_state_history: Optional[List[TrafficLightStatesDict]] = None,
+    get_reasons: bool = False,
+    get_confidence_score: bool = False,
+    get_birdviews: bool = False,
+    detect_collisions: bool = False
+) -> BlameResponse:
+    """
+    Parameters
+    ----------
+
```

Comment:
We will need to finish filling the doc string here just like drive.py, the format we use is numpy i believe

Reply:
I just copy what we have in the server side ([external.py](https://github.com/inverted-ai/driving-models/pull/803/files/ccd7a68d60e4ce1dbeb3dbf68edb8735e05b6a8c#diff-7d808c6ee389bbf93436dd303a4c8fde29cbe5992eeb27dd1d3643f757aca9daR119)) to here. Once the description there is updated, the doc here need to be updated accordingly.